specs/_features/epbs/beacon-chain.md

```diff
@@ -0,0 +1,422 @@
+    process_rewards_and_penalties(state)
+    process_registry_updates(state)
+    process_slashings(state)
+    process_eth1_data_reset(state)
+    process_effective_balance_updates(state)
+    process_slashings_reset(state)
+    process_randao_mixes_reset(state)
+    process_historical_summaries_update(state)
+    process_participation_flag_updates(state)
+    process_sync_committee_updates(state)
+    process_builder_updates(state) # [New in ePBS]
+```
+
+### Execution engine
+
+#### Request data
+
+##### New `NewInclusionListRequest`
+
+```python
+@dataclass
+class NewInclusionListRequest(object):
+    inclusion_list: InclusionList
+```
+  
+#### Engine APIs
+
+#### New `notify_new_inclusion_list`
+
+```python
+def notify_new_inclusion_list(self: ExecutionEngine,
+                              inclusion_list_request: NewInclusionListRequest) -> bool:
+    """
+    Return ``True`` if and only if the transactions in the inclusion list can be succesfully executed
+    starting from the current ``self.execution_state``, their total gas limit is less or equal that
+    ```MAX_GAS_PER_INCLUSION_LIST``, And the transactions in the list of transactions correspond to the signed summary
+    """
+    ...
+```
+
+### Block processing
+
+*Note*: the function `process_block` is modified to only process the consensus block. The full state-transition process is broken into separate functions, one to process a `BeaconBlock` and another to process a `SignedExecutionPayload`.  
+
+
+```python
+def process_block(state: BeaconState, block: BeaconBlock) -> None:
+    process_block_header(state, block)
+    process_execution_payload_header(state, block.body.execution_payload_header) # [Modified in ePBS]
+    # Removed process_withdrawal in ePBS is processed during payload processing [Modified in ePBS]
+    process_randao(state, block.body)
+    process_eth1_data(state, block.body)
+    process_operations(state, block.body)  # [Modified in ePBS]
+    process_sync_aggregate(state, block.body.sync_aggregate)
+```
+
+#### New `verify_execution_payload_header_signature`
+
+```python
+def verify_execution_payload_header_signature(state: BeaconState, signed_header: SignedExecutionPayloadHeader) -> bool:
+    # Check the signature
+    builder = state.validators[signed_header.message.builder_index]
+    signing_root = compute_signing_root(signed_header.message, get_domain(state, DOMAIN_BEACON_BUILDER))
+    return bls.Verify(builder.pubkey, signing_root, signed_header.signature)
+```
+
+#### New `process_execution_payload_header`
+
+```python
+def process_execution_payload_header(state: BeaconState, signed_header: SignedExecutionPayloadHeader) -> None:
+    assert verify_execution_payload_header_signature(state, signed_header)
+    # Check that the builder has funds to cover the bid
+    header = signed_header.message
+    builder_index = header.builder_index
+    if state.balances[builder_index] < header.value: 
+        return false
```

Comment:
Shouldn't we `assert` instead of `return false`? I'm not sure what the difference is here

Reply:
Lol this is from an earlier function